llvm/test/CodeGen/RISCV/GlobalISel/regbankselect/vec-add-sub-rv32.mir

```diff
@@ -0,0 +1,511 @@
+    ; RV32I-NEXT: [[COPY1:%[0-9]+]]:vrb(<vscale x 16 x s32>) = COPY $v11
+    ; RV32I-NEXT: [[ADD:%[0-9]+]]:vrb(<vscale x 16 x s32>) = G_ADD [[COPY]], [[COPY1]]
+    ; RV32I-NEXT: $v10 = COPY [[ADD]](<vscale x 16 x s32>)
+    ; RV32I-NEXT: PseudoRET implicit $v10
+    %0:_(<vscale x 16 x s32>) = COPY $v10
+    %1:_(<vscale x 16 x s32>) = COPY $v11
+    %2:_(<vscale x 16 x s32>) = G_ADD %0, %1
+    $v10 = COPY %2(<vscale x 16 x s32>)
+    PseudoRET implicit $v10
+
+...
+---
+name:            add_nxv1s64
+legalized:       true
+tracksRegLiveness: true
+body:             |
+  bb.0.entry:
+    liveins: $v10, $v11
+
+    ; RV32I-LABEL: name: add_nxv1s64
+    ; RV32I: liveins: $v10, $v11
+    ; RV32I-NEXT: {{  $}}
+    ; RV32I-NEXT: [[COPY:%[0-9]+]]:vrb(<vscale x 1 x s64>) = COPY $v10
+    ; RV32I-NEXT: [[COPY1:%[0-9]+]]:vrb(<vscale x 1 x s64>) = COPY $v11
+    ; RV32I-NEXT: [[ADD:%[0-9]+]]:vrb(<vscale x 1 x s64>) = G_ADD [[COPY]], [[COPY1]]
+    ; RV32I-NEXT: $v10 = COPY [[ADD]](<vscale x 1 x s64>)
+    ; RV32I-NEXT: PseudoRET implicit $v10
+    %0:_(<vscale x 1 x s64>) = COPY $v10
+    %1:_(<vscale x 1 x s64>) = COPY $v11
+    %2:_(<vscale x 1 x s64>) = G_ADD %0, %1
+    $v10 = COPY %2(<vscale x 1 x s64>)
+    PseudoRET implicit $v10
+
+...
+---
+name:            sub_nxv2s64
+legalized:       true
+tracksRegLiveness: true
+body:             |
+  bb.0.entry:
+    liveins: $v10, $v11
+
+    ; RV32I-LABEL: name: sub_nxv2s64
+    ; RV32I: liveins: $v10, $v11
+    ; RV32I-NEXT: {{  $}}
+    ; RV32I-NEXT: [[COPY:%[0-9]+]]:vrb(<vscale x 2 x s64>) = COPY $v10
+    ; RV32I-NEXT: [[COPY1:%[0-9]+]]:vrb(<vscale x 2 x s64>) = COPY $v11
+    ; RV32I-NEXT: [[SUB:%[0-9]+]]:vrb(<vscale x 2 x s64>) = G_SUB [[COPY]], [[COPY1]]
+    ; RV32I-NEXT: $v10 = COPY [[SUB]](<vscale x 2 x s64>)
+    ; RV32I-NEXT: PseudoRET implicit $v10
+    %0:_(<vscale x 2 x s64>) = COPY $v10
+    %1:_(<vscale x 2 x s64>) = COPY $v11
+    %2:_(<vscale x 2 x s64>) = G_SUB %0, %1
+    $v10 = COPY %2(<vscale x 2 x s64>)
+    PseudoRET implicit $v10
+
+...
+---
+name:            add_nxv4s64
+legalized:       true
+tracksRegLiveness: true
+body:             |
+  bb.0.entry:
+    liveins: $v10, $v11
+
+    ; RV32I-LABEL: name: add_nxv4s64
+    ; RV32I: liveins: $v10, $v11
+    ; RV32I-NEXT: {{  $}}
+    ; RV32I-NEXT: [[COPY:%[0-9]+]]:vrb(<vscale x 4 x s64>) = COPY $v10
+    ; RV32I-NEXT: [[COPY1:%[0-9]+]]:vrb(<vscale x 4 x s64>) = COPY $v11
+    ; RV32I-NEXT: [[ADD:%[0-9]+]]:vrb(<vscale x 4 x s64>) = G_ADD [[COPY]], [[COPY1]]
+    ; RV32I-NEXT: $v10 = COPY [[ADD]](<vscale x 4 x s64>)
+    ; RV32I-NEXT: PseudoRET implicit $v10
+    %0:_(<vscale x 4 x s64>) = COPY $v10
+    %1:_(<vscale x 4 x s64>) = COPY $v11
+    %2:_(<vscale x 4 x s64>) = G_ADD %0, %1
+    $v10 = COPY %2(<vscale x 4 x s64>)
+    PseudoRET implicit $v10
+
+...
+---
+name:            sub_nxv8s64
+legalized:       true
+tracksRegLiveness: true
+body:             |
+  bb.0.entry:
+    liveins: $v10, $v11
```

Comment:
Be careful which vector registers you are passing as livein. M8 operations will be using `v10, v11, v12, v13, v14,  v15, v16, v17` as a single register grouping. That means that you probably should use the next available register for the next live in.

I am calling out this specific LMUL 8 case in this test, but make sure this holds for all your test cases for their respective LMULs. It could be helpful to see which registers the IR translator for a test that lowers to this MIR test case chooses if you are not sure.

Reply:
> Be careful which vector registers you are passing as livein. M8 operations will be using `v10, v11, v12, v13, v14,  v15, v16, v17` as a single register grouping. That means that you probably should use the next available register for the next live in.
> 
> 
> 
> I am calling out this specific LMUL 8 case in this test, but make sure this holds for all your test cases for their respective LMULs. It could be helpful to see which registers the IR translator for a test that lowers to this MIR test case chooses if you are not sure.

Why hand construct the test at all? Write the IR test and use -stop-before to get the MIR.